File: leetcode/Python/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st, a, b = [], 0, 0
        for s in tokens:
            if s=='+':
                st.append(st.pop()+st.pop())
            elif s=='-':
                b, a = st.pop(), st.pop()
                st.append(a-b)
            elif s=='/':
                b, a = st.pop(), st.pop()
                st.append(int(a/b))
            elif s=='*':
                st.append(st.pop()*st.pop())
            else:
                st.append(int(s))
        return st.pop()
            

File: leetcode/C++/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        int a=0, b=0;
        for(string &s : tokens){
            if(s=="+" || s=="/" || s=="*" || s=="-"){
                b=st.top(); st.pop(); a=st.top(); st.pop();
                if(s=="+") st.push(a+b);
                else if(s=="/") st.push(a/b);
                else if(s=="*") st.push(a*b);
                else st.push(a-b);
            }
            else st.push(stoi(s));
        }
        return st.top();
    }
};